src/token_stream.rs

```diff
@@ -72,6 +72,11 @@ impl<'a> Lexer<'a> {
                 self.next();
                 TokenKind::BinaryOperator(BinaryOperator::Implication("->"))
             }
+            ('⊕', _) => TokenKind::BinaryOperator(BinaryOperator::ExclusiveDisjunction("⊕")),
+            ('⊻', _) => TokenKind::BinaryOperator(BinaryOperator::ExclusiveDisjunction("⊻")),
+            ('↮', _) => TokenKind::BinaryOperator(BinaryOperator::ExclusiveDisjunction("↮")),
+            ('≢', _) => TokenKind::BinaryOperator(BinaryOperator::ExclusiveDisjunction("≢")),
+            ('+', _) => TokenKind::BinaryOperator(BinaryOperator::ExclusiveDisjunction("+")),
```

Comment:
I don't like XOR as `+`, but i guess it makes sense since it's a binary half adder. It just kinda feels wrong.

Reply:
Do you have a suggestion for a different ascii charactor for XOR?